app/repository/AdminDataUnitLinksEditRepository.scala

```diff
@@ -1,17 +1,11 @@
 package repository
 
-import uk.gov.ons.sbr.models.{ Period, VatRef }
-import uk.gov.ons.sbr.models.edit.Patch
+import repository.rest.PatchStatus
+import uk.gov.ons.sbr.models._
 
 import scala.concurrent.Future
 
-sealed trait EditParentLinkStatus
-case object EditSuccess extends EditParentLinkStatus
-case object EditRejected extends EditParentLinkStatus
-case object EditUnitNotFound extends EditParentLinkStatus
-case object EditFailure extends EditParentLinkStatus
-case object EditConflict extends EditParentLinkStatus
-
 trait AdminDataUnitLinksEditRepository {
```

Comment:
The return type still does not feel right here - as we are leaking an implementation detail from a lower tier.  Traditionally each tier would wrap things like this, and define its own statuses / exceptions to keep the interface to each tier isolated.

One option here might simply be to rename ```PatchStatus``` to something like ```EditStatus```.  While edits are currently only done by patch, this generality could let you reuse EditStatus more widely, and avoid some of the boilerplate status mapping.

Just some thoughts at this stage.  The current approach does not feel quite right - but I don't want to delay this any longer at this point.  Better approaches may surface as you work on the sad paths.